src/class-bulk-task.php

```diff
@@ -278,18 +300,23 @@ public function filter__users_where( mixed $query ): void {
 	 *                               The callable will be passed an object of the
 	 *                               specified type.
 	 * @param string       $object_type Type of object to query.
+	 * @param ?callable    $after_batch_callback Optional. Callback to run after each batch is processed.
 	 */
-	public function run( array $args, callable $callable, string $object_type = 'wp_post' ): void {
-		if ( ! method_exists( $this, "run_{$object_type}_query" ) ) {
-			return;
-		}
-
+	public function run(
+		array $args,
+		callable $callable,
+		string $object_type = 'wp_post',
+		?callable $after_batch_callback = null,
```

Comment:
Initially, I implemented this at `Bulk_Task construct`, but then I changed my mind just because the callback is better together with the `run` method.

That however forces one to use the `wp_post` object type. I think that's okay, better to be explicit rather than clever here.